lib/chart/internal/storage/buffer.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Chart.Internal.Storage.Buffer do
+  @moduledoc false
+
+  alias Chart.Chart
+
+  @self_key :storage
+
+  def new() do
+    %{
+      callbacks: [],
+      type: __MODULE__
+    }
+  end
+
+  def add(settings) when is_map(settings) do
+    put_in(settings, [@self_key], new())
+  end
+
+  def put(%Chart{data: buffer, settings: %{storage: storage}} = chart, data) do
+    merged_data = merge(buffer, data)
+    {x_data, y_data} = merged_data |> List.flatten() |> Enum.unzip()
+
+    updated_settings =
+      storage.callbacks
+      |> Enum.zip([x_data, y_data])
+      |> Enum.reduce(chart.settings, fn {callback, data}, settings ->
+        callback.(settings, Enum.min_max(data))
+      end)
```

Comment:
Zde je otazkou zda to jde vubec udelat obecne. Nebot obecne ty callbaky nemusi byt zavisle na x/y datech. Pak je nutno rozlisit na ktere ty callbaky se maji ci nemaji aplikovat x/y data apod... Tedy uz tohle nejak zohlednit v `settings.ex` ve fci `set_callbacks`

Reply:
Nechapu, proc `callback` nedostane celou strukturu a zase ji nevrati. Tedy nejdriv ty data ulozit do `%Chart{}` a pak ho dat do `callback`. Specifikace callbacku tedy je `f(Chart.t()) -> Chart.t()`